koschei/db.py

```diff
@@ -110,6 +110,15 @@ def bulk_insert(self, objects):
                 obj.id = obj_id
             self.expire_all()
 
+    def commit_no_expire(self):
+        """
+        The same as commit, but avoids marking ORM objects as expired.
+        """
+        expire_on_commit = self.expire_on_commit
+        self.expire_on_commit = False
+        super(KoscheiDbSession, self).commit()
+        self.expire_on_commit = expire_on_commit
```

Comment:
Shouldn't this be in finally block to make sure that the flag is restored even on exception during commit?